### PR TITLE
chore: add reproducible-win

### PR DESCRIPTION
**Pull request by mndeveci**
Adds `update-reproducible-win-reqs` Makefile target as well as `requirements/reproducible-win.txt` which will be used to generate MSI installer during the release.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by hnnasit — COMMENTED** [review]
Should we also update our GH action that runs this?

**Comment by mndeveci** [comment]
> Should we also update our GH action that runs this?

Yeah I've checked that and it already runs a parent target (`update-reproducible-reqs`) which is also updated in this PR.

**Review by hnnasit — APPROVED** [review]

**Review by hawflau — APPROVED** [review]
feels like the reproducible files should be categorized by Python version instead of OS. But categorizing by OS is also fine. I guess the benefit of categorizing by OS is when we upgrade Python version in the future, it's easier for us to do it per OS.

File: requirements/reproducible-win.txt [hunk 1]
```diff
@@ -0,0 +1,785 @@
+#
```
Comment (mildaniel) [inline requirements/reproducible-win.txt:1]: Is this file different from the MacOS file? Looks like the command to create it is the same. Does it need to be run on Windows?

**Comment by mndeveci** [comment]
Abandoning this PR now until we figure out that we need to run this on a Windows environment.

File: requirements/reproducible-win.txt (hunk 1 above)
Comment (mildaniel) [reply requirements/reproducible-win.txt:1]: Also, if this should be run on a Windows machine, maybe we need to move it to a GHA?

**Comment by mndeveci** [comment]
> feels like the reproducible files should be categorized by Python version instead of OS. But categorizing by OS is also fine. I guess the benefit of categorizing by OS is when we upgrade Python version in the future, it's easier for us to do it per OS.

I thought about this one as well, especially for build images, it might be better to generate these files per python version so that we can use it seamlessly in other places.

**Pull request closed without merging**